### DISCUSSION TITLE
Ability to ignore text

### DISCUSSION THREAD

**Discussion by hasantahir** [godbout/Wooshy.docs#117] [General]
Wondering if possible to ignore the text and simply focus on the UI elements (buttons, links et al.). Understand and appreciate this may be a mess given the lack of consistency in different apps. Just to explain what I mean, sharing a screencast:

https://user-images.githubusercontent.com/11236148/220321803-09991465-6759-43fa-8dc7-3e846b1fdb7d.mov

Thanks a lot!

**Comment by godbout** [comment]
?

https://github.com/godbout/Wooshy.docs#the-art-of-searching

you can filter by buttons by using `button` (or `butt` probably), links by `link`. if you want the save button you could type `save button` or probably just `s b`.

am i missing something?

**Comment by hasantahir** [comment]
Greatly enjoy using the choices (button, link, save) but personally, it feels more natural to describe the action like `save` or `save button` or even `s b`. Can we have some sort of precedence where the text gets the least priority? In the video below, there is `save` in the two text prompts, would be great if we can directly get to the buttons/links and other UI elements before Wooshy gets to the text.

https://user-images.githubusercontent.com/11236148/220332284-bdc195aa-f197-469c-bca3-b3c75d7145dc.mov

Thanks

**Reply by godbout** [reply]
so you want to type `save` and get only the two buttons highlighted? but some people will most likely want the opposite. which is why you can mix both text, and roles. you can also add them in any order you want. you can target the `Don't save` button with `bu don` (or probably even just `don`). if you type `s bu` probably you'll get the two buttons.

**Reply by hasantahir** [reply]
Personally, I am always seeking a button/link/text box that has some actions linked to it. The video that you linked also shows Wooshy navigating to such elements. I can't think of a case where one would access just the text. 

Not that well versed by UI, but I can envisage the huge impact of this app. Like removing the need to memorise keyboard shortcuts plus using a pointing device when one can navigate likewise with Wooshy.

thanks

**Reply by godbout** [reply]
> I can't think of a case where one would access just the text.

copying text.

> Can we have some sort of precedence where the text gets the least priority?

i think it's confusing. the "precedence" that you're talking about is just the order given by macOS to the UI Elements that you would reach by using the native `tab` feature. typing `save` grabs everything that has `save`: text, links, buttons, etc. changing the "precedence" would be confusing. or at least, this would need to be an option. like you choose what roles you want to get highlighted first. but again, i can't see this as being constant. sometimes you'll want a button, sometimes you'll want text. maybe in your specific case you only use Wooshy with native apps like Word? anyways, will digest.

**Comment by godbout** [comment]
what about an option to remove anything that is not clickable. would that be enough for you? or are there times where you will want to actually reach texts.

**Reply by hasantahir** [reply]
Personally, I have used Wooshy only to reach clickable things. This option would be great for me. Thanks!

**Reply by godbout** [reply]
what about in Finder? do you use Wooshy to select files or folders? coz those are text. there's probably gonna be other places that we don't foresee that will be affected too if i add an option to ignore text.

**Comment by hasantahir** [comment]
Finder takes me to the files or folders as I type (a mix of Scrolla and Wooshy). Wooshy is handy for reaching Favourites which I think are links again.

**Reply by godbout** [reply]
you can reach anything you want in Finder without Wooshy?

**Reply by hasantahir** [reply]
Files and folders yes. Search yes. Need Wooshy for reaching Favourites and tags inside Finder. But my first step is to use Raycast to take me to the directory I want to be in.

**Reply by godbout** [reply]
i see. thanks.
